### ISSUE TITLE
hardhat transaction out of gas because of block.number

### ISSUE THREAD

**Issue by novaknole** [nomiclabs/hardhat#1303]
This is my contract.

```js
pragma solidity ^0.6.8;
pragma experimental ABIEncoderV2;

contract ACL  {
    

    uint256 a;

    constructor(address _initialRoot)  public {
         a = block.number;
    }

}
```

```js
const ACL = (await ethers.getContractFactory('ACL'))
await ACL.deploy(root)
```

After running this, I get : ` Error: Transaction ran out of gas`. 

I tried setting this as my hardhat network config.

```js
hardhat: {
      throwOnTransactionFailures: true,
      throwOnCallFailures: true,
      allowUnlimitedContractSize: true,
      blockGasLimit: 0x1fffffffffffff
    },
```

That didn't help. 

This is my `yarn.lock`. I don't use `hardhat-deploy`.

```js
hardhat@^2.0.4:
  version "2.1.0"
  resolved "https://registry.yarnpkg.com/hardhat/-/hardhat-2.1.0.tgz#4d400f1c209ea43c01316ca1864777d97de8d51b"
  integrity sha512-S3/1hM0L6DF8TjfzYlXnJ6egnWyUbB3YsuhtPBk1i4PQcgnZC9od/2uLA3Vsc6o8Fs6ZbZaAA+Ubze5uGjvmbg==
  dependencies:
    "@nomiclabs/ethereumjs-vm" "4.2.2"
    "@sentry/node" "^5.18.1"
    "@solidity-parser/parser" "^0.11.0"
    "@types/bn.js" "^4.11.5"
    "@types/lru-cache" "^5.1.0"
    abort-controller "^3.0.0"
    adm-zip "^0.4.16"
    ansi-escapes "^4.3.0"
    chalk "^2.4.2"
    chokidar "^3.4.0"
    ci-info "^2.0.0"
    debug "^4.1.1"
    enquirer "^2.3.0"
    env-paths "^2.2.0"
    eth-sig-util "^2.5.2"
    ethereum-cryptography "^0.1.2"
    ethereumjs-abi "^0.6.8"
    ethereumjs-account "^3.0.0"
    ethereumjs-block "^2.2.2"
    ethereumjs-common "^1.5.0"
    ethereumjs-tx "^2.1.2"
    ethereumjs-util "^6.2.0"
    find-up "^2.1.0"
    fp-ts "1.19.3"
    fs-extra "^7.0.1"
    glob "^7.1.3"
    immutable "^4.0.0-rc.12"
    io-ts "1.10.4"
    lodash "^4.17.11"
    merkle-patricia-tree "3.0.0"
    mnemonist "^0.38.0"
    mocha "^7.1.2"
    node-fetch "^2.6.0"
    qs "^6.7.0"
    raw-body "^2.4.1"
    resolve "1.17.0"
    semver "^6.3.0"
    slash "^3.0.0"
    solc "0.7.3"
    source-map-support "^0.5.13"
    stacktrace-parser "^0.1.10"
    "true-case-path" "^2.2.1"
    tsort "0.0.1"
    uuid "^3.3.2"
    ws "^7.2.1"

```

The problem seems to be `block.number`, because If i remove it, then it works fine. 

I try to run `hardhat compile` and `hardhat test`.

**Comment by alcuadrado** [comment]
Can you try rerunning this with `hardhat@rc`?

**Comment by novaknole** [comment]
My package.json

```
"hardhat": "rc",
```

my yarn.lock

```js
hardhat@rc:
  version "2.1.1-rc.1"
  resolved "https://registry.yarnpkg.com/hardhat/-/hardhat-2.1.1-rc.1.tgz#10891f06819bf656af882726458b0079099f07f1"
  integrity sha512-fBUFGDjsX5A6MnVhsEvO+tWlWrdVHYRGoniQdh3EaAcJsB/WbR8r2NE1Omj538O3N50/PtLbFExPaT2Psz96AA==
  dependencies:
    "@nomiclabs/ethereumjs-vm" "4.2.2"
    "@sentry/node" "^5.18.1"
    "@solidity-parser/parser" "^0.11.0"
    "@types/bn.js" "^4.11.5"
    "@types/lru-cache" "^5.1.0"
    abort-controller "^3.0.0"
    adm-zip "^0.4.16"
    ansi-escapes "^4.3.0"
    chalk "^2.4.2"
    chokidar "^3.4.0"
    ci-info "^2.0.0"
    debug "^4.1.1"
    enquirer "^2.3.0"
    env-paths "^2.2.0"
    eth-sig-util "^2.5.2"
    ethereum-cryptography "^0.1.2"
    ethereumjs-abi "^0.6.8"
    ethereumjs-account "^3.0.0"
    ethereumjs-block "^2.2.2"
    ethereumjs-common "^1.5.0"
    ethereumjs-tx "^2.1.2"
    ethereumjs-util "^6.2.0"
    find-up "^2.1.0"
    fp-ts "1.19.3"
    fs-extra "^7.0.1"
    glob "^7.1.3"
    immutable "^4.0.0-rc.12"
    io-ts "1.10.4"
    lodash "^4.17.11"
    merkle-patricia-tree "3.0.0"
    mnemonist "^0.38.0"
    mocha "^7.1.2"
    node-fetch "^2.6.0"
    qs "^6.7.0"
    raw-body "^2.4.1"
    resolve "1.17.0"
    semver "^6.3.0"
    slash "^3.0.0"
    solc "0.7.3"
    source-map-support "^0.5.13"
    stacktrace-parser "^0.1.10"
    "true-case-path" "^2.2.1"
    tsort "0.0.1"
    uuid "^3.3.2"
    ws "^7.2.1"
```

Though, same error appears.

**Comment by alcuadrado** [comment]
We continued investigating this, and the error was actually fixed in #1304. The problem was with a complex lerna setup that hoisted some packages.

We can release v2.1.1 now, @fvictorio.

**Comment by fvictorio** [comment]
We just published `hardhat@2.1.1`. Thanks @novaknole for finding this issue and helping us debug it!
